eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -56,6 +56,9 @@ jobs:
           JavaVersion: '1.11'
 
     steps:
+      - checkout: self
```

Comment:
Can this not be setup at a higher level to avoid duplication?

Reply:
There is a variable that can be set ```Agent.Source.Git.ShallowFetchDepth``` which does it globally. I'll do another PR which takes this route.